ADR/0011-roles-and-permissions.md

```diff
@@ -0,0 +1,81 @@
+# 11. Roles and Permissions for StoneSoup
+
+Date: 2023-01-10
+
+## Status
+
+Accepted
+
+## Context
+StoneSoup is using Kubernetes as the control plane for managing its resources. We require a system for managing user roles and permissions in this context. We have defined the following roles for our project: "Contributor", "Maintainer", and "Owner". We need to map these roles to specific permissions in the Kubernetes RBAC system, in terms of API groups, verbs, and resources.
+
+## Decision
+
+We will use the built-in Kubernetes RBAC system for StoneSoup's role and permissions management, and map the following roles to specific permissions, as described in the table below:
+
+### Roles
+Contributor: Members who interact with the workspace mostly through pull requests.
+Maintainer: Members who manage the workspace without access to sensitive or destructive actions.
+Owner: Members who have full access to the workspace including sensitive and destruction actions.
+
+### Roles and Permissions Table
+
+|     Role      | Permissions             | API Groups                | Verbs                             | Resources
+|---------------|-------------------------|---------------------------|-----------------------------------|----------------------------------------------------------------------
+| Contributor   | Workspace               | Access to namespaces that backs workspace                     |
+|               | Application & Component | appstudio.redhat.com      | get, list, watch                  | applications, components, componentdetectionqueries
+|               | Environment             | appstudio.redhat.com      | get, list, watch                  | promotionruns, snapshotenvironmentbindings, snapshots, environments
+|               | *GitOps*                | managed-gitops.redhat.com | get, list, watch                  | gitopsdeployments
+|               | PipelineRun             | tekton.dev                | create, get, list, watch          | pipelineruns
```

Comment:
I see that in https://docs.google.com/document/d/1KMwHbvTAJ-AVKdHo-gVq_LkgpmwKKdCtCiI04X69kII/edit# the PipelineRun is marked as available for everyone for creation. But do we really want to allow contributors to create arbitrary PipelineRuns directly? It's probably fine for now but down the road we should clarify if we want to wrap all the build APIs in a high level stone soup api so users with limited permissions like Contributors can manually initiate the build of a component (and maybe initiate a promotion?) instead of giving access to the low level and more powerful API like PipelineRuns.

Reply:
Yeah, in the doc @sbose78 mentioned:
> users shouldn't have the RBAC to create PipelineRuns directly. 

so the `create` verb should be dropped.